packages/osc-ecommerce/app/root.tsx

```diff
@@ -311,10 +323,10 @@ interface DocumentProps {
 }
 
 const Document = ({ children }: DocumentProps) => {
-    const { siteSettings, selectedLocale } = useLoaderData<typeof loader>();
+    const { siteSettings, selectedLocale, acceptedTos } = useLoaderData<typeof loader>();
     const locale = selectedLocale ?? locales.default;
-    // TODO: This probably needs to be controlled by something like a cookie banner
-    const hasUserConsent = true;
+
+    const hasUserConsent = acceptedTos;
```

Comment:
just tested the functionality using an active GTM and Hubspot ID but nothing was quite working:

May I suggest the following refactors:

```suggestion
    const hasUserConsent = acceptedTos;
    
    const showTracking =
        process.env.NODE_ENV !== 'development' &&
        (siteSettings.analytics.gtmId || siteSettings.analytics.hubspotId) &&
        acceptedTos;
```

Reply:
Not sure about this logic. We're setting showTracking to true if only one of the siteSettings ids exists. But if, for example only one existed - ie we have the gtmId and not the hubspotId, we'd still be adding the hubspot script, but with an undefined id which wouldn't work because `showTracking` would be true. I think the check would need to be on the individual level?